app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -729,6 +731,75 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml): void
         }
     }
 
+    /**
+     * Replaces strings the PHP date string parser understands with
+     * the actual time in date, datetime and hidden fields.
+     *
+     * Hidden fields must end with `|date` to prevent replacing
+     * strings by a date time representation by accident.
+     */
+    public function populateDateTimeValues(Form $form, string &$formHtml): void
+    {
+        $formName       = $form->generateFormName();
+        $fields         = $form->getFields();
+        $autoFillFields = [];
+
+        /** @var Field $field */
+        foreach ($fields as $key => $field) {
+            if (in_array($field->getType(), ['date', 'datetime', 'hidden'])) {
+                $autoFillFields[$key] = $field;
+            }
+        }
+
+        foreach ($autoFillFields as $field) {
+            $fieldType = $field->getType();
+            $token     = $field->getDefaultValue();
+
+            switch ($fieldType) {
+                case 'date':
+                    $format = 'Y-m-d';
+                    break;
+                case 'datetime':
+                case 'hidden':
+                    $format = 'Y-m-d\TH:i';
+                    break;
+
+                default:
+                    continue 2;
+            }
+
+            // prevent empty fields from getting parsed as now
+            if (empty(trim($token))) {
+                continue;
+            }
+
+            if ('hidden' === $fieldType) {
+                /* $pattern:
+                        ^([^\|]+) => match any char, except pipe
+                        \|date    => then look for `|date`
+                */
+                preg_match('/^([^\|]+)\|date$/', $token, $matches);
+
+                // prevent accidental parsing of hidden input values as date
+                if (empty($matches)) {
+                    continue;
+                }
+
+                $token = $matches[1];
+                unset($matches);
```

Comment:
Is this necessary? The array will be cleared with the garbage collector.

Reply:
Yes, some of the older code does this in Mautic. We need to address it as Tomas Votruba pointed out: https://tomasvotruba.com/blog/2020/03/02/we-do-not-need-senior-developers-we-need-senior-code-bases/